tests/response/test_file_response.py

```diff
@@ -0,0 +1,107 @@
+"""A large part of the tests in this file were adapted from:
+
+https://github.com/encode/starlette/blob/master/tests/test_responses.py And are
+meant to ensure our compatibility with their API.
+"""
+
+from pathlib import Path
+from typing import TYPE_CHECKING, AsyncIterator
+
+import anyio
+import pytest
+
+from starlite import BackgroundTask, ImproperlyConfiguredException
+from starlite.response import FileResponse
+from starlite.status_codes import HTTP_200_OK
+from starlite.testing import TestClient
+
+if TYPE_CHECKING:
+    from starlite.types import Receive, Scope, Send
+
+
+def test_file_response(tmpdir: Path) -> None:
+    path = tmpdir / "xyz"
+    content = b"<file content>" * 1000
+    Path(path).write_bytes(content)
+
+    filled_by_bg_task = ""
+
+    async def numbers(minimum: int, maximum: int) -> AsyncIterator[str]:
+        for i in range(minimum, maximum + 1):
+            yield str(i)
+            if i != maximum:
+                yield ", "
+            await anyio.sleep(0)
+
+    async def numbers_for_cleanup(start: int = 1, stop: int = 5) -> None:
+        nonlocal filled_by_bg_task
+        async for thing in numbers(start, stop):
+            filled_by_bg_task = filled_by_bg_task + thing
+
+    cleanup_task = BackgroundTask(numbers_for_cleanup, start=6, stop=9)
+
+    async def app(scope: "Scope", receive: "Receive", send: "Send") -> None:
+        response = FileResponse(path=path, filename="example.png", background=cleanup_task)
+        await response(scope, receive, send)
+
+    assert filled_by_bg_task == ""
+    client = TestClient(app)
+    response = client.get("/")
+    expected_disposition = 'attachment; filename="example.png"'
+    assert response.status_code == HTTP_200_OK
+    assert response.content == content
+    assert response.headers["content-type"] == "image/png"
+    assert response.headers["content-disposition"] == expected_disposition
+    assert "content-length" in response.headers
+    assert "last-modified" in response.headers
```

Comment:
Probably best to test the correctness of these values

Reply:
ok

Reply:
done